### PR TITLE
Focus on Service testing not just mock basics

### PR DESCRIPTION
**Pull request by sdelamo**
I've changed the guide slightly. Please @niravassar download the guide to your computer and generated the guide locally, read it and less discuss it. I changed the guide to focus on Service testing not just Mock basics. It could complement the other guide we currently have on Domain Class testing.

### CODE REVIEW

File: complete/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Fri Nov 27 23:09:32 CET 2015
+#Tue Apr 18 09:12:11 CEST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-all.zip
```
Comment (niravassar) [inline complete/gradle/wrapper/gradle-wrapper.properties:6]: just wondering how this came about, and also some of the build.gradle and gradle.properties changes. Just for my knowledge on why, etc.

File: complete/grails-app/conf/logback.groovy [hunk 2]
```diff
@@ -34,3 +34,4 @@ if (Environment.isDevelopmentMode() && targetDir != null) {
     logger("StackTrace", ERROR, ['FULL_STACKTRACE'], false)
 }
 root(ERROR, ['STDOUT'])
+logger 'grails.mock.basics', INFO, ['STDOUT'], false
```
Comment (niravassar) [inline complete/grails-app/conf/logback.groovy:37]: Does this produce some nice logging for us. just wondering.

**Review by niravassar — COMMENTED** [review]
Looks good. i am going to download it and see it end to end

File: complete/grails-app/conf/logback.groovy (hunk 2 above)
Comment (sdelamo) [reply complete/grails-app/conf/logback.groovy:37]: I replaced a print statement with a log.info statment. You will need to register a logger at level info to see that output

File: complete/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (sdelamo) [reply complete/gradle/wrapper/gradle-wrapper.properties:6]: Change from -bin.zip to -all.zip to have access to Gradle API sources

**Comment by niravassar** [comment]
Sergio, in general i see your changes and like them. I just wanted some input on how you came about this switch of focus, and then also the inclusion of the integration test. What was teh thoughts behind it in the context of the other guides. Just wanted to soak it in and talk to get a good idea.

**Comment by niravassar** [comment]
i merged the pull request into the master. I also checked in some last typo edits. It looks ready to go from my view. thanks sergio.

**Pull request merged**